### PR TITLE
Ch setup database zo67HVJ1

### PR DESCRIPTION
**Pull request by GuevaraMarius**
## What does this PR do?
-Configure the app to use the Postgres database
-Configure the app to use the sequelize ORM
## How to test it manually
- Clone this REPO
- Do `npm install` to install all packages in `package.json`
- Create `.env` file and put configs as in `.env.example`
-Create three databases: `phantom-dev` for development, `phantom-test` for testing, `phantom` for final 
> Then do **`npx sequelize-cli db:migrate `**
> Then do **`npm start`** or **`npm run dev`** to run it.
## What are relavant trello story
https://trello.com/c/zo67HVJ1/17-configure-the-app-to-use-postgres-database

### CODE REVIEW

**Review by SIBONIYO — APPROVED** [review]
good Job

**Review by mansurissa — COMMENTED** [review]
good convetions

**Review by Egide138 — APPROVED** [review]

**Review by KundwaBruno — APPROVED** [review]

File: src/database/models/user.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+const {
+  Model
+} = require('sequelize');
```
Comment (jabichris) [inline src/database/models/user.js]: Why are you using require yet we are using es6?

File: src/database/config/config.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require('dotenv').config();
```
Comment (jabichris) [inline src/database/config/config.js]: You are doing great work, but should I comment line by line to show you all the es6 conventions that you are not following?

**Review by jabichris — COMMENTED** [review]
Look at the inline comment and make sure you put `package-lock.json` in the .gitignore file.

File: src/database/config/config.js (hunk 2 above)
Comment (GuevaraMarius) [reply src/database/config/config.js]: Thanks, I have corrected conventions.

File: .env.example [hunk 3]
```diff
@@ -1 +1,4 @@
-PORT=<The port you want to run it on>
\ No newline at end of file
+PORT=<The port you want to run it on>
+DEV_DATABASE_URL=postgres://postgres:yourpassword@yourhost:yourport/database-name
+TEST_DATABASE_URL=postgres://postgres:yourpassword@yourhost:yourport/database-name
+DATABASE_URL=postgres://postgres:yourpassword@yourhost:yourport/database-name
```
Comment (ceelogre) [inline .env.example:4]: Remember that I told you this should be empty. Heroku will populate the value when in production.

File: src/database/models/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  .forEach(file => {
+    const model = sequelize.import(path.join(__dirname, file))(sequelize, Sequelize.DataTypes);
+    db[model.name] = model;
+  });
+
+Object.keys(db).forEach(modelName => {
+  if (db[modelName].associate) {
+    db[modelName].associate(db);
+  }
+});
```
Comment (ceelogre) [inline src/database/models/index.js:32]: Are you sure about the use of this?

**Review by ceelogre — APPROVED** [review]
I will approve so we can merge this not to block others but the comments should be addressed. Be sure to raise a fix after this merge.

File: src/database/models/index.js (hunk 4 above)
Comment (GuevaraMarius) [reply src/database/models/index.js:32]: They are created by default when initializing sequelize-cli, I'm not sure what to do with them

**Pull request merged**